### PR TITLE
Tobin gh690 rpci for compatibility

### PR DESCRIPTION
**Pull request by tobine**
Replace #697 
Use only VkRenderPassCreateInfo for renderpass compatibility checks and handle initial fast-path checks at callsite.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1747,12 +1747,14 @@ static bool validate_vi_against_vs_inputs(debug_report_data *report_data, VkPipe
...
+                                                    spirv_inst_iter entrypoint, VkRenderPassCreateInfo const *rpci,
+                                                    uint32_t subpass_index) {
     std::map<location_t, interface_var> outputs;
     std::map<uint32_t, VkFormat> color_attachments;
-    for (auto i = 0u; i < rp->subpassColorFormats[subpass].size(); i++) {
-        if (rp->subpassColorFormats[subpass][i] != VK_FORMAT_UNDEFINED) {
-            color_attachments[i] = rp->subpassColorFormats[subpass][i];
+    auto subpass = rpci->pSubpasses[subpass_index];
+    for (auto i = 0u; i < subpass.colorAttachmentCount; ++i) {
+        if (rpci->pAttachments[subpass.pColorAttachments[i].attachment].format != VK_FORMAT_UNDEFINED) {
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: You need to skip `VK_ATTACHMENT_UNUSED`. `rp->subpassColorFormats` already took care of that before. See #715.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2173,11 +2175,10 @@ static bool attachment_references_compatible(const uint32_t index, const VkAttac
     return false;
 }
 
-// For given primary and secondary RenderPass objects, verify that they're compatible
-static bool verify_renderpass_compatibility(const layer_data *my_data, const VkRenderPass primaryRP, const VkRenderPass secondaryRP,
-                                            string &errorMsg) {
+// For given primary RenderPass object and secondry RenderPassCreateInfo, verify that they're compatible
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: sp: secondary

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -2688,9 +2697,9 @@ static bool validate_and_capture_pipeline_shader_state(debug_report_data *report
         }
     }
 
-    if (shaders[fragment_stage] && pPipeline->renderPass) {
+    if (shaders[fragment_stage]) {
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: OK

File: layers/core_validation.cpp [hunk 4]
```diff
@@ -5733,7 +5742,7 @@ CreateGraphicsPipelines(VkDevice device, VkPipelineCache pipelineCache, uint32_t
     for (i = 0; i < count; i++) {
         pPipeNode[i] = new PIPELINE_NODE;
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: This is all a bit wobbly. We create the PIPELINE_NODE and then do a pile of validation work, mostly on its guts. Might be a good idea to replumb this in a later PR.

File: layers/core_validation.cpp [hunk 5]
```diff
@@ -2247,25 +2237,6 @@ static bool verify_renderpass_compatibility(const layer_data *my_data, const VkR
...
 }
 
-// For given primary and secondary RenderPass objects, verify that they're compatible
-static bool verify_renderpass_compatibility(const layer_data *dev_data, const VkRenderPass primaryRP,
-                                            const VkRenderPass secondaryRP, string &errorMsg) {
-    // Handle trivial match case first
-    if (primaryRP == secondaryRP)
-        return true;
-
-    auto secondary_render_pass = getRenderPass(dev_data, secondaryRP);
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: :+1:

**Comment by chrisforbes** [comment]
Need to read the last commit more thoroughly, but this looks OK.

File: layers/core_validation.cpp (hunk 1 above)
Comment (tobine) [inline layers/core_validation.cpp]: Pushed another CL to handle this

File: layers/core_validation.cpp [hunk 6]
```diff
@@ -6328,18 +6300,21 @@ BeginCommandBuffer(VkCommandBuffer commandBuffer, const VkCommandBufferBeginInfo
                         string errorString = "";
                         auto framebuffer = getFramebuffer(dev_data, pInfo->framebuffer);
                         if (framebuffer) {
-                            VkRenderPass fbRP = framebuffer->createInfo.renderPass;
-                            if (!verify_renderpass_compatibility(dev_data, fbRP, pInfo->renderPass, errorString)) {
+                            if ((framebuffer->createInfo.renderPass != pInfo->renderPass) &&
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: This does bake in the assumption that ndhandles are never reused. I think that's OK.

**Comment by chrisforbes** [comment]
:+1: good to go.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**